lib/slosilo/attr_encrypted.rb

```diff
@@ -24,9 +24,14 @@ module ClassMethods
       def attr_encrypted *a
         options = a.last.is_a?(Hash) ? a.pop : {}
         aad = options[:aad]
+
         # note nil.to_s is "", which is exactly the right thing
         auth_data = aad.respond_to?(:to_proc) ? aad.to_proc : proc{ |_| aad.to_s }
-        raise ":aad proc must take one argument" unless auth_data.arity.abs == 1 # take abs to allow *args arity, -1
+        if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.2') and aad.respond_to?(:to_proc)
+          raise ":aad proc must take two arguments" unless auth_data.arity.abs == 2 # take abs to allow *args arity, -2
+        else
+          raise ":aad proc must take one argument" unless auth_data.arity.abs == 1  # string still returns -1
+        end
```

Comment:
With the hard dependency on Ruby 3+, let's just replace the argument check and bump the gem version a major version.  This will allow us to push fixes to the the < 3.0 Ruby version while still allowing us to move forward.

Reply:
The thing with this one is that sometimes it isn't #proc, and then the arity is actually 1, so I had to keep both scenarios supported. (depends on caller)

Reply:
I checked this again and went with OR condition between arity.abs == 1 and ==2.
Without condition on Ruby version.

This depends on the caller method and on #proc itself.
@jvanderhoof